src/ut_pex.cpp

```diff
@@ -667,46 +653,47 @@ namespace libtorrent { namespace
 		if (pc.type() != peer_connection::bittorrent_connection)
 			return boost::shared_ptr<peer_plugin>();
 
-		return boost::shared_ptr<peer_plugin>(new ut_pex_peer_plugin(m_torrent
-			, *pc.native_handle(), *this));
+		bt_peer_connection* c = static_cast<bt_peer_connection*>(pc.native_handle().get());
+		auto p = boost::make_shared<ut_pex_peer_plugin>(ut_pex_peer_plugin(m_torrent, *c, *this));
```

Comment:
it's unusual to construct a temporary object and then copy it into the heap allocated one. Does it not work to pass the constructor arguments directly to make_shared? If the constructor is private, maybe you could make boost::make_shared a friend. Although, that may be tricky, since you may have to specify the exact overload with the right number of template arguments..